### PR TITLE
Hw hibernate kostevich

### PR DESCRIPTION
**Pull request by gigbat**

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -18,6 +18,24 @@
...
+            <groupId>org.hibernate</groupId>
+            <artifactId>hibernate-java8</artifactId>
+            <version>5.4.27.Final</version>
+        </dependency>
+        <dependency>
+            <groupId>mysql</groupId>
+            <artifactId>mysql-connector-java</artifactId>
+            <version>8.0.23</version>
+        </dependency>
+    </dependencies>
```
Comment (ivlytvynets) [inline pom.xml:37]: chechstyle plugin

File: src/main/java/com/dev/cinema/dao/impl/MovieDaoImpl.java [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+            session = HibernateUtil.getSessionFactory().openSession();
+            transaction = session.beginTransaction();
+            Long id = (Long) session.save(movie);
+            transaction.commit();
+            movie.setId(id);
+        } catch (Exception e) {
+            if (transaction != null) {
+                transaction.rollback();
+            }
+            throw new RuntimeException("Can't insert movie entity", e);
```
Comment (ivlytvynets) [inline src/main/java/com/dev/cinema/dao/impl/MovieDaoImpl.java]: throw custom exception

File: src/main/java/com/dev/cinema/util/HibernateUtil.java [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+
+    private HibernateUtil() {
+    }
+
+    private static SessionFactory initSessionFactory() {
+        try {
+            return new Configuration().configure()
+                    .buildSessionFactory();
+        } catch (Exception e) {
+            throw new RuntimeException("Error creating Session Factory", e);
```
Comment (ivlytvynets) [inline src/main/java/com/dev/cinema/util/HibernateUtil.java:17]: I would throw custom exception here, too. But now it's up to you

**Review by ivlytvynets — CHANGES_REQUESTED** [review]

**Review by ivlytvynets — APPROVED** [review]

File: src/main/java/com/dev/cinema/dao/impl/MovieDaoImpl.java [hunk 4]
```diff
@@ -0,0 +1,47 @@
...
+    @Override
+    public Movie add(Movie movie) {
+        Transaction transaction = null;
+        Session session = null;
+        try {
+            session = HibernateUtil.getSessionFactory().openSession();
+            transaction = session.beginTransaction();
+            Long id = (Long) session.save(movie);
+            transaction.commit();
+            movie.setId(id);
```
Comment (Sofasmile) [inline src/main/java/com/dev/cinema/dao/impl/MovieDaoImpl.java]: https://mate-academy.github.io/jv-program-fulltime/03_hibernate/checklist/hibernate_01.html

File: src/main/java/com/dev/cinema/dao/impl/MovieDaoImpl.java [hunk 5]
```diff
@@ -0,0 +1,47 @@
...
+        } catch (Exception e) {
+            if (transaction != null) {
+                transaction.rollback();
+            }
+            throw new DataProcessingException("Can't insert movie entity", e);
```
Comment (Sofasmile) [inline src/main/java/com/dev/cinema/dao/impl/MovieDaoImpl.java]:
```suggestion
            throw new DataProcessingException("Can't insert movie entity " + movie, e);
```

**Review by Sofasmile — CHANGES_REQUESTED** [review]

File: src/main/java/com/dev/cinema/dao/impl/MovieDaoImpl.java (hunk 4 above)
Comment (gigbat) [reply src/main/java/com/dev/cinema/dao/impl/MovieDaoImpl.java]: done

**Review by RomanDubovskyi — APPROVED** [review]

**Pull request merged**